core/trino-main/src/main/java/io/trino/sql/planner/plan/TableUpdateNode.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.plan;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import io.trino.metadata.TableHandle;
+import io.trino.spi.predicate.Domain;
+import io.trino.sql.planner.Symbol;
+
+import javax.annotation.concurrent.Immutable;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+@Immutable
+public class TableUpdateNode
+        extends PlanNode
+{
+    private final TableHandle target;
+    private final Symbol output;
+    private final Map<String, Domain> assignments;
```

Comment:
I use Domain as container for value which I want to update, which holds type and value itself:

```
 public static Domain singleValue(Type type, Object value, boolean nullAllowed)
    {
        return new Domain(ValueSet.of(type, value), nullAllowed);
    }
```

I see that inside we have ValueSet, so maybe I can omit Domain and just pass ValueSet.
But I choose it, because we have similar functionality when we try to pushdown predicate (we operate on Domains).

But if you think it's not needed. I can replace assignments with this:

`private final Map<String, ValueSet> assignments;`

wdyt @martint ?

Reply:
@martint As far as I see this values are travelling between nodes, so it's required to be serialisable.
If we will be using io.trino.spi.expression.Constant, we will get:

```
ava.lang.IllegalArgumentException: io.trino.server.TaskUpdateRequest could not be converted to JSON
	at io.airlift.json.JsonCodec.toJsonBytes(JsonCodec.java:214)
	at io.trino.server.remotetask.HttpRemoteTask.sendUpdate(HttpRemoteTask.java:738)
	at io.airlift.concurrent.BoundedExecutor.drainQueue(BoundedExecutor.java:80)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: No serializer found for class io.trino.spi.expression.Constant and no properties discovered to create BeanSerializer 
```

Reply:
In TableUpdateNode, they should be stored using simple Expression objects, like any other expression or value used in the plan IR. In the SPI, they should be passed as ConnectorExpression objects (since Expression is not visible to connectors)

Reply:
@martint 
I refactor TableUpdateNode to use Constant instead of Domain
https://github.com/trinodb/trino/pull/16445/commits/5d4b6180281cf912a37a5b15401c791f486cfbe6

But I am not sure about JdbcTableHandle
I tried different ways:
- if I try to use Constant - it require serialisation
- If I ignore this field in the same way as in TableUpdateNode - it becomes empty.
- If use Object it incorrectly determine int/long types.

Reply:
I see similar code when we passing QueryParameter for JdbcTableHandle:

   ```
 @JsonProperty
    public Block getValueBlock()
    {
        return nativeValueToBlock(type, value.orElse(null));
    }
```
Maybe I should go this way?